pulsar/CMakeLists.txt

```diff
@@ -66,9 +61,165 @@ add_subdirectory(system)
+
+
+##################################################################################
+# MAIN PULSAR MODULE
+##################################################################################
+
+####################
+# Compilation flags
+####################
+if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Intel")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-std=c++11")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-w3")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-restrict;-DRESTRICT=restrict")
+
+    #  2338  : switch statement does not have a default clause
+    #list(APPEND PULSAR_CORE_CXX_FLAGS "-ww2338")
+
+    # some useless remarks
+    #  1418  : external function definition with no prior declaration
+    #  1419  : external declaration in primary source file (happens in boost files)
+    #   383  : value copied to temporary, reference to temporary used (generally ok in c++)
+    #   981  : operands are evaluated in unspecified order (generally ok in c++)
+    #   869  : parameter "XXX" was never referenced.
+    #          This is due to a bug in Intel before Intel 16 that causes this
+    #          remark to be printed for variadic templates
+    # 11074  : Inlining inhibited by limit max-size (etc)
+    # 11076  : To get full report use -qopt-report=4 -qopt-report-phase ipo
+    #   444  : destructor for base class .. not virtual (even warns for classes without virtual functions)
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-wd1418")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-wd1419")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-wd383")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-wd981")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-wd869")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-wd11074")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-wd11076")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-wd444")
+elseif("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-std=c++11")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-DRESTRICT=__restrict__")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-Wall;-Wextra;-pedantic")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-Wfloat-equal;-Wshadow")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-Wswitch-default")
+    list(APPEND PULSAR_CORE_CXX_FLAGS "-Wconversion")
+
+    # maybe in the future. Seems to need a very new gcc compiler
+    #list(APPEND PULSAR_CORE_CXX_FLAGS "-Wmisleading-indentation")
+endif()
+
+
+###################
+# Other flags
+###################
+list(APPEND PULSAR_CORE_CXX_FLAGS ${OpenMP_CXX_FLAGS})
+list(APPEND PULSAR_CORE_CXX_FLAGS ${MPI_CXX_COMPILE_FLAGS})
+
+
+#################
+# Includes
+#################
+# We need to add the parent of this directory
+get_filename_component(PULSAR_CORE_PARENT_DIR ${CMAKE_SOURCE_DIR} DIRECTORY)
+list(APPEND PULSAR_CORE_CXX_INCLUDES ${PULSAR_CORE_PARENT_DIR})
+
+# External dependencies (mark as system includes)
+list(APPEND PULSAR_CORE_CXX_EXTERNAL_INCLUDES ${MEMWATCH_INCLUDE_DIR})
+list(APPEND PULSAR_CORE_CXX_EXTERNAL_INCLUDES ${Boost_INCLUDE_DIRS})
```

Comment:
Ok, I guess it was compiled boost libraries that I was remembering. Yes, everything looks fine by me. `__init__` business is perhaps a little funny, but I don't think it's flaky.